ci/check_style.sh

```diff
@@ -14,89 +14,15 @@ rapids-dependency-file-generator \
 rapids-mamba-retry env create --force -f env.yaml -n checks
 conda activate checks
 
-set +e
-
 FORMAT_FILE_URL=https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-23.02/cmake-format-rapids-cmake.json
 export RAPIDS_CMAKE_FORMAT_FILE=/tmp/rapids_cmake_ci/cmake-formats-rapids-cmake.json
 mkdir -p $(dirname ${RAPIDS_CMAKE_FORMAT_FILE})
 wget -O ${RAPIDS_CMAKE_FORMAT_FILE} ${FORMAT_FILE_URL}
 
-# TODO: enable once pre-commit checks are configured.
-# # Run pre-commit checks
-# pre-commit run --hook-stage manual --all-files --show-diff-on-failure
-
-# Run flake8 and get results/return code
-FLAKE=$(flake8 --config=python/setup.cfg)
-RETVAL=$?
-
-# Output results if failure otherwise show pass
-if [ "$FLAKE" != "" ]; then
-  echo -e "\n\n>>>> FAILED: flake8 style check; begin output\n\n"
-  echo -e "$FLAKE"
-  echo -e "\n\n>>>> FAILED: flake8 style check; end output\n\n"
-else
-  echo -e "\n\n>>>> PASSED: flake8 style check\n\n"
-fi
-
-# Check for copyright headers in the files modified currently
-COPYRIGHT=$(python ci/checks/copyright.py --git-modified-only 2>&1)
-CR_RETVAL=$?
-if [ "$RETVAL" = "0" ]; then
-  RETVAL=$CR_RETVAL
-fi
-
-# Output results if failure otherwise show pass
-if [ "$CR_RETVAL" != "0" ]; then
-  echo -e "\n\n>>>> FAILED: copyright check; begin output\n\n"
-  echo -e "$COPYRIGHT"
-  echo -e "\n\n>>>> FAILED: copyright check; end output\n\n"
-else
-  echo -e "\n\n>>>> PASSED: copyright check\n\n"
-fi
-
-# Check for a consistent #include syntax
-# TODO: keep adding more dirs as and when we update the syntax
-HASH_INCLUDE=$(python cpp/scripts/include_checker.py \
-                     cpp/bench \
-                     cpp/comms/mpi/include \
-                     cpp/comms/mpi/src \
-                     cpp/comms/std/include \
-                     cpp/comms/std/src \
-                     cpp/include \
-                     cpp/examples \
-                     cpp/src \
-                     cpp/src_prims \
-                     cpp/test \
-                     2>&1)
-HASH_RETVAL=$?
-if [ "$RETVAL" = "0" ]; then
-  RETVAL=$HASH_RETVAL
-fi
+# Run pre-commit checks
+pre-commit run --hook-stage manual --all-files --show-diff-on-failure
 
-# Output results if failure otherwise show pass
-if [ "$HASH_RETVAL" != "0" ]; then
-  echo -e "\n\n>>>> FAILED: #include check; begin output\n\n"
-  echo -e "$HASH_INCLUDE"
-  echo -e "\n\n>>>> FAILED: #include check; end output\n\n"
-else
-  echo -e "\n\n>>>> PASSED: #include check\n\n"
-fi
-
-# Check for a consistent code format
-FORMAT=$(python cpp/scripts/run-clang-format.py 2>&1)
-FORMAT_RETVAL=$?
-if [ "$RETVAL" = "0" ]; then
-  RETVAL=$FORMAT_RETVAL
-fi
-
-# Output results if failure otherwise show pass
-if [ "$FORMAT_RETVAL" != "0" ]; then
-  echo -e "\n\n>>>> FAILED: clang format check; begin output\n\n"
-  echo -e "$FORMAT"
-  echo -e "\n\n>>>> FAILED: clang format check; end output\n\n"
-else
-  echo -e "\n\n>>>> PASSED: clang format check\n\n"
-fi
+set +e
```

Comment:
The script previously always ran all checks and aggregated the exit status only at the very end. We are now left with only the pre-commit checks and clang-tidy. My intention was to maintain the general structure, but only run additional checks in case that the pre-commit checks had passed. But as you are also pointing out in your comment below, it probably makes more sense to run all checks in any case. Should this script be reduced to only run the pre-commit checks, we can drop this.

Reply:
As suggested by @bdice , I will move the execution of the `clang-tidy` check into a separate GA custom job and split this off from this PR since it is currently blocking. The implementation of said custom job is carried out in #5235 .

Reply:
Yes, the goal is to get this to be only pre-commit and then remove the `set +e`. Let pre-commit’s exit code determine the script exit code.